### ISSUE TITLE
collegecard:: prediction

### ISSUE THREAD

**Issue by ginfung** [ai-se/privacy_sharing#13]
![image](https://cloud.githubusercontent.com/assets/3222148/18621979/21c5fb78-7df8-11e6-8d66-bfa1045ebecb.png)

![school0](https://cloud.githubusercontent.com/assets/3222148/19915886/c891da0c-a08b-11e6-8e4c-019abab783fc.png)
![school1](https://cloud.githubusercontent.com/assets/3222148/19915885/c890a75e-a08b-11e6-8dbf-9bd36843e2c4.png)
![school2](https://cloud.githubusercontent.com/assets/3222148/19915884/c8907554-a08b-11e6-8dae-4f11a06ba3f2.png)
![school3](https://cloud.githubusercontent.com/assets/3222148/19915887/c892a4d2-a08b-11e6-80be-04146f5c22ca.png)

![school](https://cloud.githubusercontent.com/assets/3222148/19899133/4aa4cd58-a035-11e6-8eab-cf21f18b0856.png)

**Comment by Ginfung** [comment]
![image](https://cloud.githubusercontent.com/assets/3222148/18621979/21c5fb78-7df8-11e6-8d66-bfa1045ebecb.png)

**Comment by timm** [comment]
this is error, so LOWER values are BETTER, right?

you have 5 charts above and 1,2 and 3,4 have the same labels. how are those different?

if i asked, do we learn more from sharing that local, is that true?

if i asked, if we dont have local, but use data from others, can we build adequate predictors?

is the above saying the linear regression better than DT for this data? if so, what are the classes? like 10 breaks?  i'd then say how does DR work on fewer classes?

**Comment by ginfung** [comment]
Lower values are better.

I divided US into four regions as figure 1. (NE NW S W)

Figure 2. The training data are from schools in NE (please note the red box in fig2). If we use NE schools as training set, and applied the trained regression (or decision tree) models at the four regions test sets, we can get the RMSE error showed in fig2.

Figure3,4,5 are similar to fig2. But trained from NW S W schools respectively.

Conclusions:
1. regression tree performs similar to decision tree.
2. error in LACE1 and LACE2 are similar. But LACE2 has higher IPR #16 (fig1)
3. one CAN use data from one region to predict schools at another region. BUT
4. errors will be higher if we only use data from one region. (in the last figure, we train the model from whole country.)

**Comment by timm** [comment]
did not notice the red. now i get it. 
- have you've got a super  bellwether effect (west is best to train on? and if you do, errors elswhere are always lower?) (@rahlk : please send copy of bell wether to jianfeng)
- how do i compare these results to baseline (error with no privatization)

still think rsme is the WRONG thing to do here.

btw- nice representation of results

**Comment by ginfung** [comment]
I modified the code. The dependent variable(earnings) has become a continuous variable. so we can use RMSE here.

**Comment by ginfung** [comment]
I dont know what is bell wether. will contact rahlk

**Comment by rahlk** [comment]
Bellwether: https://github.com/rahlk/Bellwether/
See paper in doc/16Bellw.pdf

**Comment by Ginfung** [comment]
![screen shot 2016-09-26 at 10 40 39 pm](https://cloud.githubusercontent.com/assets/3222148/18858471/5455ee84-843a-11e6-9e22-4a18ebf60450.png)

**Comment by ginfung** [comment]
RMSE (unit: us dollars/yr)
FYI max value can be 140k
![screen shot 2016-11-01 at 11 50 45 pm](https://cloud.githubusercontent.com/assets/3222148/19916172/0a36fc7e-a08e-11e6-9e4c-30224923b6b2.png)

**Comment by ginfung** [comment]
![image](https://cloud.githubusercontent.com/assets/3222148/19916938/1eeedb80-a095-11e6-9d00-4a1d91d5c396.png)

Unit (RMSE/mx_value)
note that although in many cases, using data from other region might underperform training from self region, all cases have the error less than 15%.
And even training from self-region data, the error can be around 6-9%

**Comment by timm** [comment]
thanks for this. RMSE is root mean squared error?

```
   sum( 
       (actual-pred)^2
        /n 
         )**0.5
```

you seem to have less green this time. what is your color key?

## Salary results

please present these charts with a y scale log

![image](https://cloud.githubusercontent.com/assets/29195/19941479/a3154c8e-a106-11e6-815a-1effa7569e70.png)

please also present this chart with all numbers divided by max salary as percents, rounds to 1 decimal place

![image](https://cloud.githubusercontent.com/assets/29195/19941414/624fed76-a106-11e6-90d8-da2d2871e7d7.png)

## student retention results

where are they? can they be shown in this format also
?

**Comment by ginfung** [comment]
@timm color key: hollow black circle-- train the learning model basing on the self-region data (baseline)
green --  strictly better than the baseline
gray circle -- slightly worse than the baseline (within 3%)
light red -- worse than the baseline ( ~3-5%)
red -- worse than the baseline
please note that all errors are less than 16%. all are quite good!

>  please present these charts with a y scale log
> that's what the table with color points showing
> 
> student retention results
> writing the latex scripts. coming later

**Comment by timm** [comment]
results need stats : no circles should be reported for things that bootsrap and a12 say and NOT different

**Comment by ginfung** [comment]
@timm ok. I will apply the stats technique.

**Comment by Ginfung** [comment]
@timm ok. i will apply the a12 technique

**Comment by timm** [comment]
on raw data. not normalized

**Comment by ginfung** [comment]
it was not normalized. i.e. the RMSE is about 7% (*140 k) = $9800

**Comment by ginfung** [comment]
# Earnings

![image](https://cloud.githubusercontent.com/assets/3222148/19955606/2f7b10da-a159-11e6-970f-22499a9bd3a7.png)

Black point: train and test at the same region (baseline)
Null point:    A12 test said it is same or better than the baseline (marked in black point)
light red:       A12 test say it is worse than baseline by one rank
red:             A12 test say it is significant worse than the baseline

**Comment by Ginfung** [comment]
@timm from my point of view, we only failed in the last case.
 In light red cases, they were not significant worse than the baseline (around 4% worse). In such cases, using cross-region data is still valuable.

The only cases we failed is
![image](https://cloud.githubusercontent.com/assets/3222148/19955447/ed29775e-a157-11e6-9a9e-18cbb0b10df5.png)

**Comment by ginfung** [comment]
# retention rate

![image](https://cloud.githubusercontent.com/assets/3222148/20078609/42675820-a50f-11e6-9ee7-b563594c8eb2.png)

table3 W-W typo (fixed in sharelatex)
